main.js

```diff
@@ -1006,6 +1012,51 @@ class Unifi extends utils.Adapter {
         return data;
     }
 
+    /**
+     * Function to switch poe power for port of device
+     * @param {String} site
+     * @param {String} deviceMac
+     * @param {String} port
+     * @param {Boolean} val
+     */
+    async switchPoeOfPort(site, deviceMac, port, val) {
+        try {
+            this.log.info(`switchPoeOfPort: switching poe power of port ${port} for device ${deviceMac} to ${val}`);
+
+            // we have to get whole data of 'port_overrides' to change poe power of single port.
+            // we must sent the 'port_overrides' for all ports, otherwise the other port will set to default settings
+
+            const result = await this.fetchDevices(site);
+
+            const dataDevice = result.filter(x => x.mac === deviceMac);
+
+            if (dataDevice && dataDevice.length) {
+                const deviceId = dataDevice[0].device_id;
+
+                // eslint-disable-next-line prefer-const
+                let port_overrides = dataDevice[0].port_overrides;
+
+                if (port_overrides && port_overrides.length > 0) {
+                    const indexOfPort = port_overrides.findIndex(x => x.port_idx === parseInt(port));
+
+                    if (indexOfPort > 0) {
+                        port_overrides[indexOfPort].poe_mode = val ? 'auto' : 'off';
+
+                        await this.controllers[site]._request('/api/s/<SITE>/rest/device/' + deviceId.trim(), { port_overrides: port_overrides }, 'PUT');
```

Comment:
Please no direct `._request()` calls/uses! This is a private function within `node-unifi`. Please use proper methods like in https://github.com/jens-maus/node-unifi/blob/c0e7015897130d9379734cf0e855ea511ce57236/examples/poe-switch-onoff.js#L45

Reply:
didn't know this, done in da8fef7